src/main/java/bio/terra/service/dataset/DatasetDao.java

```diff
@@ -785,6 +785,31 @@ public boolean setSecureMonitoring(
     return patchSucceeded;
   }
 
+  /**
+   * Set a dataset's inherit steward flag
+   *
+   * @param id dataset UUID
+   * @param enableInheritSteward sets the inherit steward flag in the dataset
+   * @return whether the dataset record was updated
+   */
+  @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.SERIALIZABLE)
+  public boolean setInheritSteward(UUID id, boolean enableInheritSteward) {
+    String sql = "UPDATE dataset SET inherit_steward = :enabledInheritSteward WHERE id = :id";
+
+    MapSqlParameterSource params =
+        new MapSqlParameterSource()
+            .addValue("id", id)
+            .addValue("enabledInheritSteward", enableInheritSteward);
+
+    int rowsAffected = jdbcTemplate.update(sql, params);
+    boolean patchSucceeded = (rowsAffected == 1);
```

Comment:
I think the name `patch` here is a copypaste from the earlier code. Since this code doesn't mention `patch`, maybe `update would be better?
```suggestion
    boolean updateSucceeded = (rowsAffected == 1);
```